### PR TITLE
add partition filter

### PR DESCRIPTION
**Pull request by schmijos**
fixes https://github.com/coredump-ch/water-sensor-api/issues/5

### CODE REVIEW

File: test/models/measurement_test.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
-  # test "the truth" do
-  #   assert true
-  # end
+  test 'should scope measurements per sensor' do
+    create_list(:measurement, 3, sensor: create(:sensor))
+    create_list(:measurement, 3, sensor: create(:sensor))
+    create_list(:measurement, 3, sensor: create(:sensor))
+
+    measurements = Measurement.last_per_sensor(2).to_a
+    assert_equal(6, measurements.count)
```
Comment (schmijos) [inline test/models/measurement_test.rb]: Is that test good enough? Should I test ordering of the results?

File: app/controllers/measurements_controller.rb [hunk 2]
```diff
@@ -4,11 +4,14 @@ class MeasurementsController < ApplicationController
 
   # GET /measurements
   # GET /measurements.json
+  # rubocop:disable Metrics/AbcSize
   def index
     @measurements = Measurement.all
     @measurements = @measurements.where(sensor_id: params[:sensor_id].split(',')) if params[:sensor_id].present?
+    @measurements = @measurements.last_per_sensor(params[:last_per_sensor]) if params[:last_per_sensor].present?
     @measurements = @measurements.last(params[:last]) if params[:last].present?
```
Comment (dbrgn) [inline app/controllers/measurements_controller.rb]: Maybe rename `last` to `limit`, as it effectively returns the first 5 results if I'm not mistaken?

File: test/models/measurement_test.rb (hunk 1 above)
Comment (dbrgn) [inline test/models/measurement_test.rb]: It would probably be interesting to test the edge cases, e.g. what happens if there are less measurements than the limit. Or the combination of multiple filters (e.g. last_per_sensor + last).

Ordering is irrelevant in my opinion :)

**Comment by dbrgn** [comment]
@schmijos any news? :)

**Comment by rnestler** [comment]
The subject line in the commit messages should be capitalized :stuck_out_tongue_winking_eye: 
http://chris.beams.io/posts/git-commit/#capitalize

File: app/controllers/measurements_controller.rb (hunk 2 above)
Comment (schmijos) [inline app/controllers/measurements_controller.rb]: It returns the last *n* results which are the newest ones. `limit` would be like `first(n)`.

**Comment by dbrgn** [comment]
Thx!
